File: php/features.php
<?php
declare(strict_types=1);

// Jokeapi SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class JokeapiFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new JokeapiBaseFeature();
            case "test":
                return new JokeapiTestFeature();
            default:
                return new JokeapiBaseFeature();
        }
    }
}

File: py/features.py
# Jokeapi SDK feature factory

from feature.base_feature import JokeapiBaseFeature
from feature.test_feature import JokeapiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: JokeapiBaseFeature(),
        "test": lambda: JokeapiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
